cli/popper/scm.py

```diff
@@ -51,24 +61,30 @@ def get_user(debug):
     return None
 
 
-def get_ref(debug):
+def get_ref():
     """Returns the Git REF pointed by .git/HEAD"""
-    r = get_root_folder(debug)
-    cmd = "cat {}/.git/HEAD | awk '{{print $2}}'".format(r)
-    return pu.exec_cmd(cmd, debug=debug)[0]
+    r = get_root_folder()
+    return "" if repo.head.is_detached else repo.head.ref.path
 
 
-def get_sha(debug):
+def get_sha():
     """Runs git rev-parse --short HEAD and returns result"""
-    return pu.exec_cmd('git rev-parse --short HEAD', debug=debug)[0]
+    try:
+        return repo.git.rev_parse(repo.head.object.hexsha, short=True)
+    except ValueError:
+        click.secho(
+            'ERROR: Needed a single revision',
+            fg='red', bold=True, err=True, nl=False
+        )
+        sys.exit(1)
```

Comment:
we can use the `utils.fail()` method

Reply:
This is leading to circular imports

Reply:
I see. This can be resolved by having the import of `utils` within the `except` context, as explained [here](https://stackoverflow.com/a/1556429)

Reply:
I see. This can be resolved by having the import of `utils` within the `except` context, as explained [here](https://stackoverflow.com/a/1556429)